automations/python/issues_with_prs.py

```diff
@@ -40,46 +41,106 @@
+        json={"query": query},
+        headers={"Authorization": f"Bearer {get_access_token()}"},
+    )
+    if request.status_code == 200:
+        results = request.json()
+        log.debug(f"{results=}")
+        return request.json().get("data")
+    else:
+        raise Exception(
+            "Query failed to run by returning code of {}. {}".format(
+                request.status_code, query
+            )
+        )
+
+
 def get_open_issues_with_prs(
     gh: Github,
     org_handle: str,
     repo_names: list[str],
-) -> list[Issue]:
+    linked_pr_state: str,
+) -> set[tuple[str, str]]:
     """
     Retrieve open issues with linked PRs.
 
     :param gh: the GitHub client
     :param org_handle: the name of the org in which to look for issues
     :param repo_names: the name of the repos in which to look for issues
-    :return: the list of open issues with linked PRs
+    :param linked_pr_state: the state of the linked PRs to filter by
+    :return: a set of tuples containing the repo name and issue number for each issue
     """
 
-    all_issues = []
+    all_issues = set()
     for repo_name in repo_names:
-        log.info(f"Looking for issues with PRs in {org_handle}/{repo_name}")
-        issues = gh.search_issues(
-            query="",
-            sort="updated",
-            order="desc",
-            **{
-                "repo": f"{org_handle}/{repo_name}",
-                "is": "issue",
-                "state": "open",
-                "linked": "pr",
-            },
+        log.info(
+            f"Looking for {linked_pr_state} PRs in {org_handle}/{repo_name} "
+            f"with linked issues"
+        )
+        results = run_query(
+            """
+{
+    repository(owner: "%s", name: "%s") {
+        pullRequests(first: 100, states:%s,
+                     orderBy:{field:UPDATED_AT, direction:DESC}) {
+            nodes {
+                number
+                title
+                closingIssuesReferences (first: 50) {
+                    edges {
+                        node {
+                            number
+                            title
+                            state
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+        """
+            % (org_handle, repo_name, linked_pr_state.upper())
         )
-        issues = list(issues)
+        pulls = results["repository"]["pullRequests"]["nodes"]
+        log.info(f"Found {len(pulls)} PRs in {org_handle}/{repo_name}")
+        issues = {
+            issue["node"]["number"]: issue["node"]["title"]
+            for pull in pulls
+            for issue in pull["closingIssuesReferences"]["edges"]
+        }
         log.info(f"Found {len(issues)} issues")
-        for issue in issues:
-            log.info(f"• #{issue.number} | {issue.title}")
-        all_issues += issues
+        for number, title in issues.items():
+            log.info(f"• #{number} | {title}")
```

Comment:
```suggestion
            log.info(f"• #{number: >5} | {title}")
```
Nitpicking, would look nicer in logs if the PR numbers and `|` sign are aligned

Reply:
Works for me!